File: tournaments/computeDefiniteIntegral/computeDefiniteIntegral.js
function computeDefiniteIntegral(l, r, p) {
    let result = 0;
    let a = l;
    let b = r;

    for (let i = 0; p.length > i; ++i) {
        result += (p[i] * (b - a) / (i + 1));
        a *= l;
        b *= r;
    }

    return result;
}

File: tournaments/computeDefiniteIntegral/computeDefiniteIntegral.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub computeDefiniteIntegral {
    my ($l, $r, $p) = @_;

    my $result = 0;
    my $left = $l;
    my $right = $r;
    my $len = @$p;

    for (my $i = 0; $len > $i; ++$i) {
        $result += (@$p[$i] * ($right - $left) / ($i + 1));
        $left *= $l;
        $right *= $r;
    }

    return $result;
}
